### DISCUSSION TITLE
Tfsec id map tp trivy id?

### DISCUSSION THREAD

**Discussion by sarjuymd** [aquasecurity/tfsec#2116] [Help and suppports]
Hi. 

Is there a map available of tfsec named ID's to trivy ID's?

For example:

tfsec -  aws-iam-no-policy-wildcards
trivy - AVD-AWS-0057. (https://avd.aquasec.com/misconfig/aws/iam/avd-aws-0057/)

I have over a hundred rules to re-evaluate and doing it manually is going to be incredibly time consuming and potentially a blocker for me to migrate to trivy.

**Comment by adilburaksen** [comment]
No official map document is published, but there's a reliable way to do this at scale.

The tfsec and trivy checks all live in the same place now — the `aquasecurity/defsec` codebase that powers both tools. Each check defines its legacy tfsec ID and its current AVD ID in the same rule file:

```go
Rule: scan.Rule{
    AVDID:      "AVD-AWS-0057",
    LegacyID:   "aws-iam-no-policy-wildcards",
    ...
}
```

**Quickest path for bulk migration:**

Clone or browse https://github.com/aquasecurity/defsec/tree/master/rules/cloud and `grep` for your tfsec IDs:

```sh
git clone --depth=1 https://github.com/aquasecurity/defsec
grep -r "aws-iam-no-policy-wildcards" defsec/rules/
# → shows the file containing AVDID: "AVD-AWS-0057"
```

Then you can generate the full map in one shot:

```sh
cd defsec
grep -r 'AVDID\|LegacyID' rules/ | grep -A1 'LegacyID' |   awk '/LegacyID/{id=$NF} /AVDID/{print id, $NF}' | tr -d '",' | sort
```

That gives you a `tfsec-id → avd-id` table for every check in the repo.

**Alternative: AVD website search**

https://avd.aquasec.com/misconfig/ — each check page shows both identifiers. You can also filter by provider (AWS/GCP/Azure) and the URL slug matches the AVD ID, so it's scriptable.

The `tfsec --print-checks` output also shows the AVD link next to each check name, which is useful if you only need to map a subset.
